### PR TITLE
STAR-CCM block

### PR DESCRIPTION
**Pull request by orbsmiv**
Based upon the easyblock in this upstream PR:
https://github.com/easybuilders/easybuild-easyblocks/pull/1613

### CODE REVIEW

File: easybuild/easyblocks/s/star_ccm.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            (_, ec) = run_cmd(cmd, simple=False, log_all=False, log_ok=False)
+            if ec != 0:
+                if self.starccm_subdir is None or self.starview_subdir is None:
+                    self.find_starccm_subdirs()
+                logfilepaths = glob.glob(os.path.join(
+                    self.installdir,
+                    self.starccm_subdir,
+                    "STAR-CCM+%s*_InstallLog.log" % self.version
+                )
+                )
```
Comment (edmondac) [inline easybuild/easyblocks/s/star_ccm.py]: I'd put both of these close brackets on line 83, after `self.version`

File: easybuild/easyblocks/s/star_ccm.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+            cmd = ' '.join([
+                install_script,
+                "-i silent",
+                "-DINSTALLDIR=%s" % self.installdir,
+                "-DINSTALLFLEX=false",
+                "-DADDSYSTEMPATH=false",
+                "-DPRODUCTEXCELLENCEPROGRAM=0",
+            ])
+            # Run the command, ignoring the exit code
```
Comment (edmondac) [inline easybuild/easyblocks/s/star_ccm.py]: You say "ignoring", but then you do things if ec!=0 . Perhaps "# Run the command, which may well have a non-zero exit code even though it has succeeded"

File: easybuild/easyblocks/s/star_ccm.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+            (_, ec) = run_cmd(cmd, simple=False, log_all=False, log_ok=False)
+            if ec != 0:
```
Comment (edmondac) [inline easybuild/easyblocks/s/star_ccm.py:77]: Is that really what you mean? Presumably if someone managed to get it to exit with 0, you would still want it to do what's in this block? In which case just remove the if and un-indent the block.

File: easybuild/easyblocks/s/star_ccm.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+                if self.starccm_subdir is None or self.starview_subdir is None:
```
Comment (edmondac) [inline easybuild/easyblocks/s/star_ccm.py:79]:
```suggestion
                # The script may well have exited with non-fatal errors. We'll check.
                if self.starccm_subdir is None or self.starview_subdir is None:
```

**Review by edmondac — CHANGES_REQUESTED** [review]
see comments - mostly good though

File: easybuild/easyblocks/s/star_ccm.py (hunk 1 above)
Comment (orbsmiv) [reply easybuild/easyblocks/s/star_ccm.py]: Resolved by d6a27ee6

File: easybuild/easyblocks/s/star_ccm.py (hunk 2 above)
Comment (orbsmiv) [reply easybuild/easyblocks/s/star_ccm.py]: Resolved in 7870d566

**Review by edmondac — APPROVED** [review]
lgtm

**Pull request merged**